src/main/scala/org/monarchinitiative/dosdp/cli/Generate.scala

```diff
@@ -77,12 +77,19 @@ object Generate {
         dataListVar <- dataListVars.keys
         filler <- row.get(dataListVar).flatMap(stripToOption)
       } yield dataListVar -> MultiValue(filler.split(DOSDP.MultiValueDelimiter).map(_.trim).to(Set))).toMap
+      val internalVarBindings = (for {
+        internalVars <- dosdp.internal_vars.toSeq
+        internalVar <- internalVars
+        function <- internalVar.apply.toSeq
+        value = function.apply(Some(dataListBindings.getOrElse(internalVar.input, listVarBindings.getOrElse(internalVar.input, MultiValue(Set.empty[String])))))
+        if value.isDefined
+      } yield internalVar.var_name ->  SingleValue(value.get) ).toMap
```

Comment:
Instead, avoid `isDefined` and `get` (I didn't test this):

```scala
val internalVarBindings = (for {
        internalVars <- dosdp.internal_vars.toSeq
        internalVar <- internalVars
        function <- internalVar.apply.toSeq
        value <- function.apply(Some(dataListBindings.getOrElse(internalVar.input, listVarBindings.getOrElse(internalVar.input, MultiValue(Set.empty[String])))))
      } yield internalVar.var_name ->  SingleValue(value) ).toMap
```

Reply:
Recommended logic worked perfectly, changed accordingly.